manim/mobject/types/vectorized_mobject.py

```diff
@@ -1017,6 +1041,50 @@ def point_from_proportion(self, alpha: float) -> np.ndarray:
 
             current_length += length
 
+    def proportion_from_point(
+        self,
+        point: typing.Iterable[typing.Union[float, int]],
+    ) -> float:
+        """Returns the proportion along the path of the :class:`VMobject`
+        a particular given point is at.
+
+        Parameters
+        ----------
+        point
+            The Cartesian coordinates of the point which may or may not lie on the :class:`VMobject`
+
+        Returns
+        -------
+        float
+            The proportion along the path of the :class:`VMobject`.
```

Comment:
Here you should describe what happens when the point is off the bezier path (at different scales)

Reply:
Could you elaborate a bit? I mentioned in the `Raises` part that if the point doesn't lie on the curve, a `ValueError` is raised.

Reply:
What is the tolerance for error, and what if someone wishes to change that tolerance?

(This may be a much more complicated request than I think it will be; if you think it'll take more than, say, 10 lines of code, and it works for many values in testing, then fixing it is too complex for this PR)

Reply:
I'm not sure I follow. The tolerance for point equality is (unfortunately) hardcoded to be 10<sup>-6</sup> for all VMobjects, and following this, the default number of decimal places to round all coordinates to in `proportions_along_bezier_curve_for_point` is the number of decimal places in 10<sup>-6</sup>.

So if the point is within 10<sup>-6</sup> MUnits of the curve, it's considered to be on the curve. I haven't given a way to change this in this function, well, since all VMobjects are "supposed" to use 10e-6 as their tolerance for point related stuff.

Reply:
Totally understand, thank you for doing the digging with respect to that. Looks good to me!